026-service-account-patching.md

```diff
@@ -0,0 +1,73 @@
+# Service Account patching
+
+This proposal suggests to introduce proper reconciliation of service accounts and handle them in the same way as other Kubernetes resources.
+
+## Current situation
+
+Strimzi currently does not reconcile service accounts it creates.
+In the service account reconciliation loop, only creation and deletion actually does something.
+Patching just returns success without doing anything.
+This was originally done this way because the patching removed the attached secret and was causing a new authentication token being created every reconciliation.
+
+## Motivation
+
+For many resources we create, we offer customization using the template mechanism.
+User can use the `template` sections in the different custom resources to customize some parts of the Kubernetes resources.
+In some case, it allows customizing only labels and annotations.
+In other there are more things to configure.
+One of the advantages of this model is that users can declaratively configure things in the custom resources.
+
+However, there is currently no template for service accounts.
+And even if we add the template, it will work only at creation time because we do not patch service accounts right now
+So that would create a weird situation where the template for service accounts behaves differently from other resources. 
+
+At the same time, it is in many cases desired to have the labels and annotations configurable.
+Some platforms - such as AWS - link their internal identities to the service accounts based on annotations.
+For example, the annotation `eks.amazonaws.com/role-arn: arn:aws:iam::<AWS_ACCOUNT_ID>:role/<IAM_ROLE_NAME>` tells AWS that pods with this service account should have the role specified in the annotation (For more info, see [AWS documentation](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts-technical-overview.html)).
+This is especially useful for something like Kafka Connect with connectors interacting with other AWS services.
+The connectors can use the AWS IAM role assumed by the annotation to authenticate against AWS services.
+
+## Proposal
+
+This proposal suggests several changes to how the Strimzi Cluster Operator handles the service accounts.
+
+1) A new `serviceAccount` fields will be added to the `template` sections of the Strimzi CRDs.
```

Comment:
By "Strimzi CRDs" you mean all of them? `Kafka`, `KafkaConnect` and so on?

Reply:
Well, the ones which cause any service accounts to be created. I made it a bit more clear.

Reply:
Thanks Jakub 👍